hooks/pre-command

```diff
@@ -29,31 +29,37 @@ fi
 # around between invocations; by default we persist content-addressed directories such as
 # packages, artifacts and compiled `.ji` files
 if [[ "${BUILDKITE_PLUGIN_JULIA_ISOLATED_DEPOT:-true}" == "true" ]]; then
-    # Export `JULIA_DEPOT_PATH` so that future julia invocations use this depot
-    export JULIA_DEPOT_PATH="${CACHE_DIR}/depots/${BUILDKITE_PIPELINE_ID}"
+    # Store the depot path for internal script use
+    export DEPOT_DIR="${CACHE_DIR}/depots/${BUILDKITE_PIPELINE_ID}"
+    # Export `JULIA_DEPOT_PATH` with trailing separator so Julia can use its shipped pkgimage caches
+    if [[ "$OSTYPE" == "msys" ]]; then
```

Comment:
This method of OS detection seems a bit fragile to me

Does `"msys"` means that Bash has to come from msys2 to detect Windows properly?

Reply:
It's what we've been doing before 🤷 
https://github.com/JuliaCI/julia-buildkite-plugin/blob/1b3c741abae5f05308e448c9a2cd6719fa882ea8/hooks/pre-command#L194-L215